src/backend/marsha/development/management/commands/load_development_datasets.py

```diff
@@ -0,0 +1,298 @@
+        """
+        Add arguments to management command:
+         - allow to run command without dropping the database existing data.
+        """
+        parser.add_argument(
+            "--flush-db",
+            action="store_true",
+            help="Remove all data in database before creating development dataset",
+            default=False,
+        )
+
+    @atomic
+    def handle(self, *args, **options):
+        """Execute management command."""
+
+        if options["flush_db"]:
+            self.stdout.write("Flushing database...")
+            call_command(
+                "flush",
+                verbosity=3,
+                interactive=False,
+                reset_sequences=True,
+                allow_cascade=True,
+                inhibit_post_migrate=True,
+            )
+            self.stdout.write(" - done.")
+
+        self._create_site()
+        self._create_superuser()
+        self._create_users()
+        localhost_cs = self._create_consumer_sites()
+        organization = self._create_organizations(localhost_cs)
+        self._create_playlists(localhost_cs, organization)
+
+    def _create_site(self):
+        Site.objects.get_or_create(domain="localhost:8060", defaults={"name": "Marsha"})
+
+    def _create_superuser(self):
+        """Create a superuser, required for the command."""
+        self.stdout.write("Creating superuser...")
+
+        try:
+            user = User.objects.get(username=self.admin_username)
+            self.stdout.write(" - superuser already exists.")
+        except User.DoesNotExist:
+            user = User.objects.create_superuser(
+                username=self.admin_username,
+                email=f"{self.admin_username}@example.com",
+                password=self.admin_username,
+            )
+            self.stdout.write(" - done.")
+
+        self.superuser = user  # pylint: disable=attribute-defined-outside-init
+
+    def _create_users(self):
+        """Create users, required for the command"""
+        for username in (
+            "consumer_site_admin",
+            "consumer_site_instructor",
+            "consumer_site_student",
+            "organization_admin",
+            "organization_instructor",
+            "organization_student",
+            "playlist_admin",
+            "playlist_instructor",
+            "playlist_student",
+            "instructor",
+        ):
+            self.stdout.write(f"Creating {username}...")
+
+            try:
+                user = User.objects.get(username=username)
+                self.stdout.write(f" - {username} already exists.")
+            except User.DoesNotExist:
+                user = User.objects.create_user(
+                    username=username,
+                    email=f"{username}@example.com",
+                    password=username,
+                )
+                self.stdout.write(" - done.")
+
+            setattr(self, username, user)  # warning: magic here
```

Comment:
Can you explain the magic ?

Reply:
I added a comment in code, it is simply to emphasize the fact we create here some class properties used later in other methods.